transmogrifier/models.py

```diff
@@ -290,6 +300,9 @@ class TimdexRecord:
     summary: list[str] | None = field(
         default=None, converter=dedupe, validator=optional(list_of(str))
     )
+    timdex_provenance: TimdexProvenance | None = field(
+        default=None, validator=optional(instance_of(TimdexProvenance))
+    )
```

Comment:
Why is this field optional? 🤔 In what cases is this field set to `None`?

Reply:
Bad answer: if it were a "v1" record, it wouldn't have it.  So this simplifies backwards cmpatibility.

Better answer: it allows for creating the `TimdexRecord` incrementally.  You don't have to _initialize_ a record with it set, which allows for setting that property/field more dynamically as the object moves through it's init and handling.  Even better, we somehow validate required fields during serialization.  But doesn't feel that critical.